File: app/src/main/java/com/nilhcem/blenamebadge/device/model/Mode.kt
package com.nilhcem.blenamebadge.device.model

import androidx.annotation.StringRes
import com.nilhcem.blenamebadge.R

enum class Mode(val hexValue: Byte, @StringRes val stringResId: Int) {
    LEFT(0x00, R.string.mode_left),
    RIGHT(0x01, R.string.mode_right),
    UP(0x02, R.string.mode_up),
    DOWN(0x03, R.string.mode_down),
    FIXED(0x04, R.string.mode_fixed),
    SNOWFLAKE(0x05, R.string.mode_snowflake),
    PICTURE(0x06, R.string.mode_picture),
    ANIMATION(0x07, R.string.mode_animation),
    LASER(0x08, R.string.mode_laser)
}

File: lib/bademagic_module/models/mode.dart
enum Mode {
  left('0x00'),
  right('0x01'),
  up('0x02'),
  down('0x03'),
  fixed('0x04'),
  snowflake('0x05'),
  picture('0x06'),
  animation('0x07'),
  laser('0x08');

  final String hexValue;
  const Mode(this.hexValue);

  // Helper method to safely parse hex value
  static Mode fromHex(String hexValue) {
    return Mode.values.firstWhere(
      (mode) => mode.hexValue == hexValue,
      orElse: () => Mode.left, // Default to Mode.left if no match
    );
  }
}
